### PR TITLE
Add basic tests

### PR DESCRIPTION
**Pull request by megheaiulian**
Add basic item creation and select/deselect tests

### CODE REVIEW

File: cosmoz-grouped-list.js [hunk 1]
```diff
@@ -388,19 +388,10 @@
...
-			this.data.some(function (group) {
-				var found = group.items.indexOf(item) > -1;
-				if (found) {
-					foundGroup = group;
-				}
-				return found;
-			});
-
-			return foundGroup;
+			return this._groupsMap && this.data.find(function (group) {
```
Comment (nomego) [inline cosmoz-grouped-list.js]: The reason for the old implementation is that we haven't been able to move to ES6 yet.
Neither through transpiling or platform support adjustments.
Otherwise Array.find() would have been a given.
This implementation should have been an Array.filter().find() though in my opinion for readability.

File: cosmoz-grouped-list.js [hunk 2]
```diff
@@ -526,19 +513,17 @@
...
-			if (this._groupsMap) {
-
-				this.data.forEach(function (group) {
-					var groupState = this._groupsMap.get(group);
-					groupState.selected = true;
-					this.splice.apply(this, ['selectedItems', this.selectedItems.length - 1, 0].concat(group.items));
-				}, this);
-			} else {
-				this.splice.apply(this, ['selectedItems', 0, 0].concat(this.data));
+			if(groups) {
```
Comment (nomego) [inline cosmoz-grouped-list.js]: Space between `if` and conditions - `if (groups)`. Present throughout but only commenting here.
Should be alerted by ESLint I think?

File: cosmoz-grouped-list.js [hunk 3]
```diff
@@ -526,19 +513,17 @@
...
+				selected = selected.reduce(function (all, group) {
+					var state = groups.get(group);
+					if(state) state.selected = true;
```
Comment (nomego) [inline cosmoz-grouped-list.js]: Never shorthand statements, always enclose in:
 ```
if (cond) {
  ...
}
```

File: cosmoz-grouped-list.js [hunk 4]
```diff
@@ -526,19 +513,17 @@
...
+					return all.concat(group.items || []);
+				}, []);
 			}
+			this.splice.apply(this, ['selectedItems', 0, this.selectedItems.length -1 ].concat(selected));
```
Comment (nomego) [inline cosmoz-grouped-list.js]: spacing - `this.selectedItems.length - 1]`

File: demo/basic.html [hunk 5]
```diff
@@ -0,0 +1,149 @@
+<!doctype html>
```
Comment (nomego) [inline demo/basic.html]: As far as I know, all demos on one page and iron-demo-helpers / demo-snippet is the best way to go right now and most compatible with webcomponents.org.

See https://www.webcomponents.org/element/PolymerElements/paper-button/demo/demo/index.html for example

File: demo/basic.html [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+							notify: true,
+						},
+						selectedItems: {
+							type: Array
+						},
+					},
+
+					attached: function () {
+						this.async(function () {
+							var items = [
```
Comment (nomego) [inline demo/basic.html]: This object structure is too spaced out, should be along the lines of:

```js
items: [{
  name: 'name',
  id: 'id'
}, {
  name: 'name2'
  id: 'id2'
}]
```

Also `items` isn't needed.

Any reason why this isn't just set on the `data` property's `value: function () { return.. }` ?

File: test/basic.html [hunk 7]
```diff
@@ -36,36 +13,252 @@
...
-			<cosmoz-grouped-list></cosmoz-grouped-list>
+			<cosmoz-grouped-list>
+				<cosmoz-grouped-list-template id="itemTemplate">
+					<template>
+						<div class="item-template" style="border-bottom: 1px solid grey;">
+							<div>
+								ID:<span class="item-id">{{ item.id }}</span>
+								NAME:<span class="item-name">{{ item.name }}</span>
+								VALUE:<span class="item-value">{{ item.value }}</span>
+								<div>
```
Comment (nomego) [inline test/basic.html]: looks like this should be `</div>` with less indent?

Same for group template

File: test/basic.html [hunk 8]
```diff
@@ -36,36 +13,252 @@
...
 	</test-fixture>
 
 	<script>
-	/*global WCT, suite, fixture, setup, test, Polymer, assert, logStore, errorStore */
 	(function () {
 		'use strict';
 
 		suite('<cosmoz-grouped-list>', function () {
-			var myElement;
+			var element
```
Comment (nomego) [inline test/basic.html]: comma at end

File: test/basic.html [hunk 9]
```diff
@@ -36,36 +13,252 @@
...
+			, items;
+
+			setup(function (done) {
+				element = fixture('cosmoz-grouped-list-fixture');
+				items = [
```
Comment (nomego) [inline test/basic.html]: Same object spacing as above

Comment (nomego) [inline test/basic.html]: Should probably be called `data` or `groups` or just referred to as `element.data` to make test readability better instead of `items[i].items[o]`

File: demo/basic.html (hunk 6 above)
Comment (megheaiulian) [inline demo/basic.html]: I use this only for manual testing to figure out the expected result of tests.

File: cosmoz-grouped-list.js (hunk 2 above)
Comment (megheaiulian) [inline cosmoz-grouped-list.js]: There was not eslint rule for that. I've added http://eslint.org/docs/rules/keyword-spacing

File: cosmoz-grouped-list.js (hunk 3 above)
Comment (megheaiulian) [inline cosmoz-grouped-list.js]: There was no eslint rule for that. Will add `"curly":["error","all"]`  as per http://eslint.org/docs/rules/curly.

File: cosmoz-grouped-list.js (hunk 4 above)
Comment (megheaiulian) [inline cosmoz-grouped-list.js]: There was no eslint rule for that. Will add `space-infix-ops` http://eslint.org/docs/rules/space-infix-ops rule

File: demo/basic.html (hunk 6 above)
Comment (megheaiulian) [inline demo/basic.html]: Will set this up in property's definition `value`.

Comment (megheaiulian) [inline demo/basic.html]: Added `array-bracket-spacing` rule to enforce this.

File: test/basic.html (hunk 7 above)
Comment (megheaiulian) [inline test/basic.html]: Fixed

File: test/basic.html (hunk 8 above)
Comment (megheaiulian) [inline test/basic.html]: No eslint rule for that. Will add [comma-style](http://eslint.org/docs/rules/comma-style) rule to enforce it.

File: test/basic.html (hunk 9 above)
Comment (megheaiulian) [inline test/basic.html]: Changed items to groups

File: demo/basic.html (hunk 5 above)
Comment (megheaiulian) [inline demo/basic.html]: Can we do this in a different pull request? There is a bit more work involved in reorganizing demos.

File: cosmoz-grouped-list.js (hunk 1 above)
Comment (megheaiulian) [inline cosmoz-grouped-list.js]: Reverted to old implementation with fixes.

File: demo/basic.html (hunk 5 above)
Comment (nomego) [inline demo/basic.html]: Sounds good
